src/main/java/io/antmedia/streamsource/StreamFetcherManager.java

```diff
@@ -89,7 +90,12 @@ public void setRestartStreamFetcherPeriod(int restartStreamFetcherPeriod) {
 
 
 	public StreamFetcher startStreaming(Broadcast broadcast) {	
-
+		IResourceMonitor monitor = (IResourceMonitor) scope.getContext().getBean(IResourceMonitor.BEAN_NAME);
+		int cpuLoad = monitor.getAvgCpuUsage();
+		if(cpuLoad > monitor.getCpuLimit()) {
+			logger.error("Stream Fetcher can not be created due to high cpu load: {}", cpuLoad);
```

Comment:
It's good to write stream id as well so that we understand which stream is not started